Common/Securities/SecurityService.cs

```diff
@@ -130,7 +133,12 @@ public class SecurityService : ISecurityService
             switch (symbol.ID.SecurityType)
             {
                 case SecurityType.Equity:
-                    security = new Equity.Equity(symbol, exchangeHours, quoteCash, symbolProperties, _cashBook, _registeredTypes, cache);
+                    var primaryExchange = PrimaryExchange.UNKNOWN;
+                    if (_primaryExchangeProvider?.GetPrimaryExchange(symbol.ID) != null)
+                    {
+                        primaryExchange = (PrimaryExchange)Enum.Parse(typeof(PrimaryExchange), _primaryExchangeProvider.GetPrimaryExchange(symbol.ID));
```

Comment:
Suggest we push this logic into the `_primaryExchangeProvider` implementation instead of returning `null`, will make for a better usage 👍